bee-node/build.rs

```diff
@@ -36,3 +89,60 @@ fn main() -> Result<(), BuildError> {
 
     Ok(())
 }
+
+#[cfg(feature = "dashboard")]
+mod dashboard {
+    use super::*;
+
+    use sha2::{Digest, Sha256};
+    use zip::ZipArchive;
+
+    use std::{
+        io::{self, BufReader, Cursor},
+        path::Path,
+    };
+
+    const RELEASE_URL: &str =
+        "https://github.com/iotaledger/node-dashboard/releases/download/v1.0.0/node-dashboard-bee-1.0.0.zip";
+    const RELEASE_CHECKSUM: &str = "d61bce7d70b51ea88536b22c5341ccdc3c4651bf8e02e093a771312f3f046c30";
+
+    pub(super) fn fetch<P: AsRef<Path>>(dashboard_dir: P) -> Result<(), BuildError> {
+        println!("downloading latest dashboard release from {}", RELEASE_URL);
+
+        let client = reqwest::blocking::Client::builder()
+            .user_agent("bee-fetch-dashboard")
+            .build()
+            .expect("could not create client");
+
+        let mut archive = tempfile::tempfile().expect("could not create temp file");
+
+        let response = client
+            .get(RELEASE_URL)
+            .send()
+            .and_then(|resp| resp.error_for_status())
+            .map_err(|e| BuildError::DashboardRequest(e.status().map(|code| code.as_u16()), RELEASE_URL.to_string()))?;
+
+        let content = response.bytes().map_err(|_| BuildError::DashboardDownload)?;
+
+        let mut sha256 = Sha256::new();
+        sha256.update(&content);
+        let checksum = format!("{:x}", sha256.finalize());
+
+        if checksum != RELEASE_CHECKSUM {
+            return Err(BuildError::DashboardInvalidChecksum);
+        }
+
+        println!("checksum ok");
+
+        let mut content = Cursor::new(content);
+        io::copy(&mut content, &mut archive).expect("copying failed");
```

Comment:
Like, is there no crate that does a checksum on a file ? You could do it here

Reply:
Done :+1: